astroquery/esa/jwst/core.py

```diff
@@ -0,0 +1,1231 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+=======================
+eJWST Astroquery Module
+=======================
+
+European Space Astronomy Centre (ESAC)
+European Space Agency (ESA)
+
+"""
+import binascii
+import gzip
+import os
+import shutil
+import tarfile
+import zipfile
+from urllib.parse import urlencode, quote_plus
+from builtins import isinstance
+from datetime import datetime
+
+from astropy import log
+from astropy import units
+from astropy.coordinates import SkyCoord
+from astropy.table import vstack
+from astropy.units import Quantity
+from astroquery.exceptions import RemoteServiceError
+from requests.exceptions import ConnectionError
+from astroquery.utils import send_request
+
+from astroquery.query import BaseQuery
+from astroquery.ipac.ned import Ned
+from astroquery.simbad import Simbad
+from astroquery.utils import commons
+from astroquery.utils.tap import TapPlus
+from astroquery.vizier import Vizier
+from . import conf
+from .data_access import JwstDataHandler
+
+
+__all__ = ['Jwst', 'JwstClass']
+
+
+class JwstClass(BaseQuery):
+
+    """
+    Proxy class to default TapPlus object (pointing to JWST Archive)
+    """
+
+    JWST_DEFAULT_COLUMNS = ['observationid', 'calibrationlevel', 'public',
+                            'dataproducttype', 'instrument_name',
+                            'energy_bandpassname', 'target_name', 'target_ra',
+                            'target_dec', 'position_bounds_center',
+                            'position_bounds_spoly']
+
+    PLANE_DATAPRODUCT_TYPES = ['image', 'cube', 'measurements', 'spectrum']
+    ARTIFACT_PRODUCT_TYPES = ['info', 'thumbnail', 'auxiliary', 'science',
+                              'preview']
+    INSTRUMENT_NAMES = ['NIRISS', 'NIRSPEC', 'NIRCAM', 'MIRI', 'FGS']
+    TARGET_RESOLVERS = ['ALL', 'SIMBAD', 'NED', 'VIZIER']
+    CAL_LEVELS = ['ALL', 1, 2, 3, -1]
+    REQUESTED_OBSERVATION_ID = "Missing required argument: 'observation_id'"
+
+    def __init__(self, tap_plus_handler=None, data_handler=None):
```

Comment:
Maybe make `tap_plus_handler` and `data_handler` kwarg only, too?

Reply:
Done